assembl/static2/js/app/components/cookies/cookiesSelectorContainer.jsx

```diff
@@ -0,0 +1,162 @@
+// @flow
+import React from 'react';
+import { compose, graphql } from 'react-apollo';
+import { connect } from 'react-redux';
+
+import CookiesSelector from './cookiesSelector';
+import { getCookieItem, setCookieItem } from '../../utils/globalFunctions';
+
+// graphql
+import acceptedCookiesQuery from '../../graphql/acceptedCookiesQuery.graphql';
+import updateAcceptedCookies from '../../graphql/mutations/updateAcceptedCookies.graphql';
+import withLoadingIndicator from '../common/withLoadingIndicator';
+
+import type { CookieObject } from './cookieToggle';
+import { COOKIE_TRANSLATION_KEYS } from '../../constants';
+
+type Props = {
+  updateAcceptedCookies: Function,
+  cookiesList: Array<string>
+};
+
+export type CookiesObject = {
+  other?: Array<CookieObject>,
+  essential?: Array<CookieObject>,
+  analytics?: Array<CookieObject>
+};
+
+type State = {
+  activeKey: ?string,
+  show: boolean,
+  cookies: ?CookiesObject
+};
+
+class CookiesSelectorContainer extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    const cookiesFromBrowser = getCookieItem('cookies_configuration');
+    const cookiesList = props.cookiesList ?
+      // if the user is logged in, we get the cookiesList from the query
+      props.cookiesList
+      // otherwise, we take it from the browser
+      : cookiesFromBrowser && cookiesFromBrowser.split(',');
+
+    const cookiesArray = cookiesList && cookiesList.map(
+      cookie => ({ ...this.getCookieObject(cookie), accepted: this.isCookieAccepted(cookie), cookieType: cookie })
+    );
+    const cookiesByCategory = cookiesArray && this.getCookiesObjectFromArray(cookiesArray);
+
+    this.state = {
+      activeKey: 'essential',
+      show: true,
+      cookies: cookiesByCategory
+    };
+  }
+
+  isCookieAccepted = (cookie: string) => {
+    if (cookie.startsWith('ACCEPT')) {
+      return true;
+    }
+    return false;
+  }
+
+  getCookieObject = (cookie: string) => {
```

Comment:
it returns a `CookieObject`, right?

Reply:
Actually just part of the object, I renamed it